app/src/test/java/cm/aptoide/pt/app/AppViewManagerTest.java

```diff
@@ -0,0 +1,717 @@
+    when(flagManager.flagApk("anyString", "anyString", voteType.name()
+        .toLowerCase())).thenReturn(Single.just(genericResponseV2));
+    //And the response from the request has errors
+    when(genericResponseV2.isOk()).thenReturn(false);
+    when(genericResponseV2.hasErrors()).thenReturn(true);
+
+    //Then it should inform the presenter that something went wrong
+    appViewManager.flagApk("anyString", "anyString", FlagsVote.VoteType.GOOD)
+        .test()
+        .assertValue(false);
+
+    //And the flagManager method to flag the apk should be called
+    verify(flagManager).flagApk("anyString", "anyString", voteType.name()
+        .toLowerCase());
+  }
+
+  @Test public void subscribeStoreTest() {
+    //When the presenter asks to subscribe a store
+    appViewManager.subscribeStore("anyString")
+        .test()
+        .assertCompleted();
+
+    //Then a method from storeUtilsProxy to subscribe a store should be called
+    verify(storeUtilsProxy).subscribeStore("anyString", null, null, aptoideAccountManager);
+  }
+
+  @Test public void showRootInstallWarningPopupTest() {
+    //When the presenter asks to if it should show a warning popup, and it's supposed to show
+    when(installManager.showWarning()).thenReturn(true);
+    //Then it should return true
+    Assert.assertEquals(true, appViewManager.showRootInstallWarningPopup());
+  }
+
+  @Test public void saveRootInstallWarningTest() {
+    //When the presenter asks to save a rootInstallWaring
+    appViewManager.saveRootInstallWarning(true);
+    //Then a method from the installManager should be called
+    verify(installManager).rootInstallAllowed(true);
+  }
+
+  @Test public void downloadAppTest() {
+    //Cache App (Test preparation)
+    DetailedApp detailedApp =
+        new DetailedApp((long) 1, "any", "packageName", (long) 1, "any", "any", "any", "any", true,
+            null, null, null, null, null, (long) 1, "", null, null, 1, null, null, store, null,
+            appStats, null, null, null, true, true, "");
+
+    DetailedAppRequestResult detailedAppRequestResult = new DetailedAppRequestResult(detailedApp);
+
+    AppViewConfiguration appViewConfiguration =
+        new AppViewConfiguration((long) 1, "packageName", "anyString", "", null, null, "", "", 0.0,
+            "", "");
+
+    appViewManager =
+        new AppViewManager(installManager, downloadFactory, appCenter, reviewsManager, adsManager,
+            storeManager, flagManager, storeUtilsProxy, aptoideAccountManager, appViewConfiguration,
+            preferencesManager, downloadStateParser, appViewAnalytics, notificationAnalytics,
+            installAnalytics, limit, socialRepository);
+
+    when(appCenter.loadDetailedApp((long) 1, "anyString", "packageName")).thenReturn(
+        Single.just(detailedAppRequestResult));
+    when(store.getId()).thenReturn((long) 1);
+    when(storeManager.isSubscribed(anyLong())).thenReturn(Observable.just(true));
+
+    appViewManager.loadAppViewViewModel()
```

Comment:
Same as above. If this is not being saved or used, there is not need to call toBlocking and value

Reply:
Agree